manticore/core/cpu/abstractcpu.py

```diff
@@ -416,15 +428,15 @@ def canonicalize_instruction_name(self, instruction):
         pass
 
     def execute(self):
-        ''' Decode, and execute one instruction pointed by register PC'''
-        if not isinstance(self.PC, (int,long)):
-            raise SymbolicPCException()
+        '''
+        Decode, and execute one instruction pointed by register PC
+        '''
 
-        if not self.memory.access_ok(self.PC,'x'):
-            raise InvalidPCException(self.PC)
+        # Decode the instruction if it wasn't explicitly decoded
+        if self._decoded_pc != self.PC:
```

Comment:
yeah, this shouldn't happen. this can be an assert, i'm just protecting against someone doing `cpu.execute()` without explicitly decoding first.